### PR TITLE
move SRW/TIMER/STRING_UTILS to c shared

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: tests/string_utils_int/block_storage_string_utils_int.c [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+#endif
+
+#include "windows.h"
+
+#include "testrunnerswitcher.h"
+
+#include "block_storage_string_utils.h"
+
+
+BEGIN_TEST_SUITE(block_storage_string_utils_int_tests)
```
Comment (dcristoloveanu) [inline tests/string_utils_int/block_storage_string_utils_int.c]: >block_storage_ [](start = 17, length = 14)

Seems to be string_utils

Comment (anporumb) [reply tests/string_utils_int/block_storage_string_utils_int.c]: Changed!

---
In reply to: [281848300](https://github.com/Azure/azure-c-shared-utility/pull/309#discussion_r281848300) [](ancestors = 281848300)

File: CMakeLists.txt [hunk 2]
```diff
@@ -324,12 +327,14 @@ set(source_h_files
...
     ./inc/azure_c_shared_utility/sha.h
     ./inc/azure_c_shared_utility/socketio.h
+    ./inc/azure_c_shared_utility/srw_lock.h
     ./inc/azure_c_shared_utility/stdint_ce6.h
     ./inc/azure_c_shared_utility/strings.h
     ./inc/azure_c_shared_utility/strings_types.h
     ./inc/azure_c_shared_utility/string_token.h
     ./inc/azure_c_shared_utility/string_tokenizer.h
     ./inc/azure_c_shared_utility/string_tokenizer_types.h
+    ./inc/azure_c_shared_utility/string_utils.h
```
Comment (jebrando) [inline CMakeLists.txt]: >./inc/azure_c_shared_utility/string_utils.h [](start = 4, length = 43)

You forgot timer.h in the list.  Is that intentional.

File: adapters/timer.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        (void)QueryPerformanceCounter(&(result->startTime)); /*from MSDN:  On systems that run Windows XP or later, the function will always succeed and will thus never return zero.*/
+    }
+    return result;
+}
+
+void timer_start(TIMER_HANDLE timer)
+{
+    if (timer == NULL)
+    {
+        LogError("invalid arg TIMER_HANDLE timer=%p", timer);
```
Comment (jebrando) [inline adapters/timer.c]: >=%p", timer [](start = 48, length = 11)

Do you really need this evaluation.  Just hard code NULL in here.

**Pull request merged**